### ISSUE TITLE
Next steps?

### ISSUE THREAD

**Issue by eddelbuettel** [ess-intro/ess-intro#7]
Shall we become "more formal" and aim for slides + videos for the first few presentations?  Some are clearly ready and we should probably aim towards a date using the old "deadline device".

/cc for comments:  @aezarebski @plantarum @frederic-santos @chr1swallace @greg-minshall and of course everybody else who is "lurking"

**Comment by frederic-santos** [comment]
Yep, I'm almost ready to begin! (Just procrastinating because I'm not proud of my french accent :smile:)
I can push an updated version of the slides and record a video by the end of the month.

**Comment by aezarebski** [comment]
Even the prospect of a deadline is having the intended effect!

**Comment by plantarum** [comment]
Hi,

I will try to put a video together this weekend. If anyone would 
like to contribute a peer review for my RMarkdown tutorial, please 
do so!

Best,

Tyler


Dirk Eddelbuettel writes:

> Shall we become "more formal" and aim for slides + videos of the 
> first few videos? Some are clearly ready and we should
> probably aim towards a date using the old "deadline device".
>
> /cc for comments: @aezarebski @plantarum @frederic-santos 
> @chr1swallace @greg-minshall and of course everybody else who
> is "lurking"
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.


-- 
Tyler Smith
plantarum.ca

**Comment by eddelbuettel** [comment]
We don't have to rush for "tomorrow" or "this weekend" -- I just wanted to feel the "overall temperature".  It looks to me that we have a set of four that are near ready so we should help each other getting those ready and maybe "publish" those (how to do that is still a puzzle: I only every put YouTube videos up under my handle -- should we create one for the ESS Project?).

Happy to help with videos if anybody needs help.  Happy OBS user, have let it loose on my students for a few runs of the one class I teach too...

**Comment by aezarebski** [comment]
Has the temperature dropped below freezing?

**Comment by eddelbuettel** [comment]
Shall we poll who is ready for videos, and then record those?

**Comment by frederic-santos** [comment]
Yup. Still have a bunch of little fixes to do before recording the video, but I'm _almost_ ready.

**Comment by aezarebski** [comment]
Mine is pretty much ready, but no idea what to do re recordings, so probably going to lurk until there is an example to copy :)

**Comment by eddelbuettel** [comment]
Install [Open Broadcaster Software](https://obsproject.com/), then fight the settings a little / poke at tutorials / ask here. I have been using it for a few years now to record short lecture videos and mini-tutorials as for the shell series preceding this.  It's good and featureful, if quirky in the UI.

That said, I don't think anybody has given me feedback on my short (too short?) basic editing intro...

**Comment by frederic-santos** [comment]
Last questions: where should we host the videos? Are we free to choose a (potentially) different host for each tutorial (Youtube, Vimeo, etc.); or should we host all videos within one single Youtube/Vimeo/whatever playlist for instance?

If we pick the "one single playlist for all tutos" option, how can we handle that? (Who will create the account and/or upload the videos, etc.?)

**Comment by eddelbuettel** [comment]
I wondered about that too.  A single account would be nicer.  One proxy is to copy all into an existing account. 

Any volunteers to create a 'joint' new YouTube handle for us?

**Comment by frederic-santos** [comment]
I recorded my video this morning, it's ready for upload when the account is created!

**Comment by eddelbuettel** [comment]
According to a quick Google search, one can create different 'brand accounts' so I could move forward and create, say 'ess-intro' and then make everybody ready to upload a co-manager of the brand.  Sounds like plan?   Any better, catchier, more on target names than 'ess-intro' or is that it?  It is what we use here too...

**Comment by aezarebski** [comment]
"ESS super series", how can you beat a recursive acronym which also has a name clash?

**Comment by frederic-santos** [comment]
> According to a quick Google search, one can create different 'brand accounts' so I could move forward and create, say 'ess-intro' 

My video begins with a "Welcome to the ESS-intro blah blah...", so I'm comfortable with that :smile:

**Comment by eddelbuettel** [comment]
Nice :)  I used a recursive scheme for a few packages too (`binb`, `linl`, `pinp`, `tint`, `dang`, ...).  Let's if there if there are any even more clever ideas.

**Comment by greg-minshall** [comment]
hi, Dirk,

(trying to come out of a ditch...)

obs is cool!  thanks.

> That said, I don't think anybody has given me feedback on my short
> (too short?) basic editing intro...

i seem to have missed this.  where is this?  was it a video?  or, a
comment in one of the issues?  i'm not sure what to look for.

if it doesn't include this: do you have any off-the-cuff usage, best
(or, "pretty good") practice hints?  what sorts of "scripts" tend to
work, etc.?  do you do much post-processing, or try to just trim the
start up cruft at the beginning (and at the end)?

cheers, Greg

**Comment by eddelbuettel** [comment]
Sorry, poor wording: "editing" for ESS, here at https://github.com/ess-intro/presentation-first-steps and indexed in our table at https://github.com/ess-intro/ess-intro/wiki/Topics

I have never edited a single second of OBS.  Record, and post.  Or ditch and re-cord.

**Comment by eddelbuettel** [comment]
Sorry, something got in the way but I have been meaning to
 - create that new 'brand account' for us, and while @aezarebski makes a clever point I think I will stick with 'ESS Intro' for consistency with what we have here
 - record mine as a first pass too and upload
 - we can then hopefully collect a few more over the few weeks (besides @frederic-santos who has one too already, I think) and then announce.
 
Sounds like a plan?   Give me some :+1: or :-1: below ...

**Comment by aezarebski** [comment]
Solid plan. 

In the broader next steps vibe, I don't really understand the "online presence" thing, but perhaps we should consider a twitter post/R-bloggers thing once a couple of recordings are ready.

**Comment by eddelbuettel** [comment]
Made my initial recording and am uploading now.  A shareable link from YouTube:  https://youtu.be/RDUQIx5LvnU
Corresponding pdf slides at https://github.com/ess-intro/presentation-first-steps/blob/master/first-steps.pdf

To add other videos, I can make whoever has one a co-manager of our 'branded' (YouTube's language, not mine) channel 'Emacs Speaks Statistics'.  Ping me here (or via private email if you don't want to share a mail handle here).  

PS Now also realized to redo the pdf in non-incremental handout mode.

**Comment by farcego** [comment]
Great work.
I watched in the morning in my office and I found the sound volume in the low spectrum. But from home. It sounds ok. I often have issues with sound volume, specially with recordings in English, so it is likely a personal think and not an issue of the video.

**Comment by frederic-santos** [comment]
Great video, thanks Dirk @eddelbuettel !



> To add other videos, I can make whoever has one a co-manager of our 'branded' (YouTube's language, not mine) channel 'Emacs Speaks Statistics'. Ping me here (or via private email if you don't want to share a mail handle here).

My email: frederic.santos [at] u-bordeaux.fr
My recording will be longer than yours (~ 11 minutes, I couldn't do better) and you'll have to put up with an awful french accent, but I hope this video will be useful though ;)

**Comment by eddelbuettel** [comment]
Invite (to Fred) sent!  Let's see if you can add it.

**Comment by eddelbuettel** [comment]
@frederic-santos Any luck uploading your file? I do not see anything under channel content (apart from the one I uploaded).

**Comment by plantarum** [comment]
Started working on my first video last night. Having some audio/visual issues: I look and sound like a weirdo. I'm afraid this may not be a technical issue, but is in fact a too-accurate reflection of reality.

**Comment by eddelbuettel** [comment]
You just need to overcome that barrier.  Hits me each time.  But then random strangers will tell you they liked the video and may even have found it helpful...

**Comment by frederic-santos** [comment]
> @frederic-santos Any luck uploading your file? I do not see anything under channel content (apart from the one I uploaded).

I've just uploaded the video, but still as a private (or unlisted) video. If you think that everything is okay, we can make it public.

**Comment by eddelbuettel** [comment]
Well if I cannot see it no other team can, so I woukd suggest listing it.  We're simply not just publicising them (apart from talking here).  Mine seems to have gotten 17 views; given that there are about as many of us here I am not too concerned.

**Comment by eddelbuettel** [comment]
Ah sorry -- did you _just_ upload it?  Now I see it so that is fine.  If you prefer unlisted we can keep it that way.

**Comment by eddelbuettel** [comment]
@frederic-santos I _really_ liked that video (and learned a thing or two, of course).  I think it's ready too.

**Comment by frederic-santos** [comment]
Many thanks!
I'll thus set the video to public visibility: just wanted to be sure that it was good enough before doing that.

**Comment by alexey0308** [comment]
@frederic-santos The public audience finds your video really great!

**Comment by greg-minshall** [comment]
Dirk,

nice.  i did also notice the sound level to be low on my laptop.  on my
tablet, i could get the sound level up.  but, then, when the video
finished and an ad was shown, its sound level was very high.
Frédéric's, on the other hand, has great sound levels.

i learned about the key combination C-M-\.

and, i remembered one of my favorite things about emacs is the paste
buffer.  that one can hit C-y, then *immediately* M-y (>= 1 times) M-y,
and get a history of previous yanks, i use often, every day.

Frédéric, i learned a lot from your presentation.

one suggestion, though, is to use `M-x COMMAND` rather than customized
key strokes.  i thought of that in your nice discussion of r-dired.

i liked the presence of the indicators for mouse and keyboard modifiers
in the upper right hand corner.  is that part of "Open Broadcaster
Software"?

cheers, Greg

**Comment by frederic-santos** [comment]
Thanks Alexey and Greg for your comments!

> i liked the presence of the indicators for mouse and keyboard modifiers
> in the upper right hand corner.  is that part of "Open Broadcaster
> Software"?

Nope, that's [key-mon](https://github.com/scottkirkwood/key-mon) :) I must admit that I stole the idea from [Rainer's screencasts about Org mode](https://www.youtube.com/watch?v=sQS06Qjnkcc): this helps a lot for Emacs tutorials.

**Comment by eddelbuettel** [comment]
Thanks for the comments from me too.  My standalone microphone was not on the preferred setting, but computer and obs were all at the 'max' so that may be as good as I can do (short of re-recording and "shouting" more which I may just have to do...).

**Comment by alexey0308** [comment]
Hi Dirk, @eddelbuettel maybe one can try [ffmpeg](https://trac.ffmpeg.org/wiki/AudioVolume) to increase the volume?

**Comment by eddelbuettel** [comment]
_Very_ intriguing option. The page lists wav files, would that work with wav-hidden-inside-mp4?

[goes trying -- apparently!]

Worked.  That is frigging fabulous.  I tried the 'times 1.5', will try 'times 2.0' and see how to replace the mp4 at YT. (Apparently one cannot replace so we will have mine twice, once as a "silent movie" :) and once as a louder action movie.  

Did something similar for my students once, this will be very helpful.  Thanks **a bunch** for the hint!   Linux still rocks harder than anything else if one manages to find the right rocks :)

**Comment by alexey0308** [comment]
It should work, but I am only afraid, that the background noise can be also multiplied.

**Comment by eddelbuettel** [comment]
But lesson learned ... and after two mins with obs documentation now added a 5db volume increase via the audio 'gain' filter. Yay.

**Comment by greg-minshall** [comment]
Frédéric,


> Nope, that's [key-mon](https://github.com/scottkirkwood/key-mon) :) I
> must admit that I stole the idea from [Rainer's screencasts about Org
> mode](https://www.youtube.com/watch?v=sQS06Qjnkcc): this helps a lot
> for Emacs tutorials.

thanks.  i'll try key-mon.  (any idea, anyone, how to get it to
recognize a re-mapped CAPS LOCK key [to CTRL]?)

and, thanks for the link to Rainer's screencasts!  i remembered someone
mentioned some videos, but hadn't been able to track them down.
serendipity strikes again!

cheers, Greg

**Comment by eddelbuettel** [comment]
A wiki page with a video resources may be a good idea.  Did I just see @greg-minshall volunteering for that?  :wink: 

Any of us, really, could or should start that....

**Comment by greg-minshall** [comment]
hmm, on volunteering.  i was going to include the link, with a few others, i guess, in my rumored-to-be presentation.  where would that

about which presentation, a question.  probably, because if comes naturally to me, i would like to provide some org mode files and a makefile, which would churn on those, and turn the whole mess into a whole mess of .html and .pdf files.

on gitlab, i do this with gitlab's CI/CD (whatever that is), and the results go on "gitlab pages" (in my case, something like `minshall.gitlab.io/blahblahblah`).

how would something like that fit into people's concept of a reasonable world?  (i assume github has some similar facility.)

one alternative is that i do the make on my own machine, and the output gets committed in the git repository.  i find that slightly distasteful (but, i've certainly done much more distasteful things).

**Comment by eddelbuettel** [comment]
Minimally viable setups please. Render your pdf from your input locally, commit inputs and pdf.  We're not here to showcase the wild wild west of varying build setups.  We're here to provide an intro to ESS.

**Comment by greg-minshall** [comment]
Dirk,

> Minimally viable setups please. Render your pdf from your input
> locally, commit inputs and pdf.  We're not here to showcase the wild
> wild west of varying build setups.  We're here to provide an intro to
> ESS.

and here i wanted to be known as the Wyatt Earp of ESS intros... :)

that makes sense.  thanks.

**Comment by greg-minshall** [comment]
> A wiki page with a video resources may be a good idea. Did I just see @greg-minshall volunteering for that? 
> 
> Any of us, really, could or should start that....

does this "subtree of repositories" admit to a top-level wiki page?  if not, is there an obvious place to put such a list?

**Comment by eddelbuettel** [comment]
See https://github.com/ess-intro/ess-intro/wiki which we had for a few weeks.  I thought this was common knowledge...

**Comment by aezarebski** [comment]
Now there is there web page, that would probably be a suitable place highlight the wiki etc

**Comment by eddelbuettel** [comment]
Good point.  Added it in https://github.com/ess-intro/ess-intro.github.io/commit/c34b8593614accb9a3a4f4fa27bdc1491be9d9b0

**Comment by greg-minshall** [comment]
[wiki videos](https://github.com/ess-intro/ess-intro/wiki/Related-Videos)

**Comment by alexey0308** [comment]
[this series](https://www.youtube.com/c/SystemCrafters/videos) about Emacs is very informative and I find it worth adding as well.

**Comment by greg-minshall** [comment]
Alexey, sure, i'll add those (sight unseen).

**Comment by eddelbuettel** [comment]
@greg-minshall Sorry, but I have to revert that, You committed to the top-level, had you read previous messages more carefully you might have noticed that we collect these _to the side in a wiki page_.

Moreover, you also committed _to the wrong repo_.  The web page repo would have been here: https://github.com/ess-intro/ess-intro.github.io

If in doubt, just ask.  Or make it a pull request.  Thank you.

**Comment by greg-minshall** [comment]
Dirk,

sorry.  i noticed i was to put it on the wiki after putting it *one*
wrong place.  then i thought i had deleted that.  (the old gray matter
ain't what she used to be...)  and, then i put it in the wiki, but i
guess the wrong one?
----
https://github.com/ess-intro/ess-intro/wiki/Related-Videos

(am i right, wiki pages aren't defined in our git repositories?  we just
edit them online?)

Greg

**Comment by eddelbuettel** [comment]
Folks: please see #10 as well and comment there.

**Comment by greg-minshall** [comment]
are we targetting ESS version 18 (18.10, or some such)?  i.e., rather than the unreleased (but, i guess, enhanced) 19.

**Comment by eddelbuettel** [comment]
I would think that may be up to each presenter -- we also did not require an Emacs version (everybody "naturally" seems to be on 27.1 or later) or OS version or ...

But if and when it matters, make it clear which one it is.

**Comment by plantarum** [comment]
Underway. I'm nearly down to 10 minutes. However, the sound quality is objectively poor, very quiet, lots of static and background noise. I'll try to secure a better microphone tomorrow.

I'll definitely have *something* for Thursday, but probably just the base-install video.

**Comment by greg-minshall** [comment]
Dirk, et al., i did a first recording yesterday.  i had an incorrect emacs key-sequence on screen, so i'll re-do it.

but, i'm worried about length.  i was basically happy with my delivery, but it came in at 20 minutes.  the re-do will presumably come in less (having done it once), but probably not significantly less.

it's not a lot of content, but coupled with my discursive presentation style, i guess the minutes add up.

let me know what you think.

**Comment by plantarum** [comment]
I have the same problem. My first run through was >20 minutes. After a few practices I'm down to 11, but have dropped a lot of detail from my written tutorial. 

I think it's useful to go through the process of identifying the small number of essential concepts that you need to convey. And it may be valuable to separate ideas that are best expressed in a video, and those that are more appropriate written down?

For my RMarkdown tutorial, I'm thinking about using the video to live-demo writing an RMarkdown document, with the configuration details left to the written document.

Although, at the moment my main issue is figuring out how to filter out the noise in my audio, which is surprisingly loud.

--
plantarum.ca

**Comment by eddelbuettel** [comment]
@greg-minshall I would be preferable if you could cut it to (well) below 10 minutes.

**Comment by greg-minshall** [comment]
@eddelbuettel okay.  i'll try.

**Comment by eddelbuettel** [comment]
On the margin three focussed 10 min videos >> one long 30 min video.  Or maybe get 5 x 8 with proper intros than one 30-ish one.

**Comment by greg-minshall** [comment]
@plantarum 

`For my RMarkdown tutorial, I'm thinking about using the video to live-demo writing an RMarkdown document, with the configuration details left to the written document.`

thanks for the suggestion.  i'll try this.  (when was that deadline again? :)

**Comment by greg-minshall** [comment]
@eddelbuettel Dirk, i apologize, but even with the code development approach,i don't seem to be able to whittle below 20 minutes.  maybe too many academic talks of that length.

i don't have a youtube/google login.  i've put the .mkv file [here](http://minshall.org/code-2021-04-15-08-20-17.mkv).  could i ask you to add it to the youtube site?  the slide deck is located [here](https://ess-intro.github.io/presentation-org-mode/artefacts/ess-org-beamer.pdf), or i can (i assume) add it to the github.io site if you prefer.

**Comment by eddelbuettel** [comment]
I believe youtube requires mp4 format.  Can you please convert it?

**Comment by greg-minshall** [comment]
@eddelbuettel will (try to) do.  i'll also shrink -- during my "pandemic walk" this morning i thought of how i should trim it.  it's down to 14 minutes.  details to follow.

**Comment by eddelbuettel** [comment]
Go for another walk and shrink it below ten.  The 'academic' excuse is not even an excuse for an excuse as we had 5 minute lightning talks for a decade.  If you need a mental model, think recording a poster.  Or two to three talks.

**Comment by greg-minshall** [comment]
a walk a day...

**Comment by greg-minshall** [comment]
[mp4](http://minshall.org/writing-code-2021-04-15-15-25-33.mp4)

**Comment by eddelbuettel** [comment]
Uploaded, and updated wiki to make it clearer that this was a one off.  Authors _own_ their content, and this covers rights and responsibilities including uploads.

**Comment by aezarebski** [comment]
For future readers, there are instructions for adding your recordings [here](https://github.com/ess-intro/ess-intro/wiki/Infrastructure). If you find these instructions unclear, please help the next person by editing them :)

**Comment by eddelbuettel** [comment]
I did :)

**Comment by aezarebski** [comment]
@greg-minshall once there is an available link to the recording, can you please make an edit to [this](https://github.com/ess-intro/ess-intro.github.io/blob/main/index.Rmd) file to include it on the website.

Sorry, making you the guinea pig to see how easy it is for others to edit this...

**Comment by eddelbuettel** [comment]
(Not the be a stickler but the file had three authors committing to it so he is hardly the first.)  I was planning to update it unless someone beats me to it as I am hoping to get a (short) blog post, tweet, email to ess-help, ... out this evening.   

Nice work all, we got four videos and slide decks in!

**Comment by plantarum** [comment]
I thought I sent my youtube account to @eddelbuettel via email last night, but it looks like it went to some github notification account instead?

if someone can add me to the youtube account, it I'm ready to upload my installation video. My user name is sedgeboy @ gmail

**Comment by eddelbuettel** [comment]
Well I do not know here you sent it too by my email addresses aren't exactly hidden. I'll add you so that an invite to you goes out. [Done]

**Comment by plantarum** [comment]
I replied to one of the messages that github sent me with your name in the "from" box. When you didn't respond I checked and noticed that the actual email address looked odd.

**Comment by plantarum** [comment]
Alrighty then. I'm now a youtuber. D'you suppose I'll get any endorsement contracts before the weekend?

**Comment by eddelbuettel** [comment]
I would expect the Eclipse, Atom and VS Code teams to engage in a bidding war for your services :)

**Comment by greg-minshall** [comment]
@eddelbuettel thanks for uploading!

@aezarebski the (self-imposed) problem for me is that i don't have youtube or gmail or google credentials.

**Comment by aezarebski** [comment]
Urgh! I'm embarrassed to say, that possibility had not even occurred to me. Probably not even that uncommon among emacs users...
